contracts/seed/Seed.sol

```diff
@@ -187,53 +155,57 @@ contract Seed {
             isFunded = true;
         }
         // fundingAmount is an amount of fundingTokens required to buy _seedAmount of SeedTokens
-        uint256 seedAmount = (_fundingAmount*PCT_BASE)/price;
-
-        // Funding Token balance of this contract;
-        uint256 fundingBalance = fundingCollected;
+        uint256 seedAmount = (_fundingAmount*PRECISION)/price;
 
         // feeAmount is an amount of fee we are going to get in seedTokens
-        uint256 feeAmount = (seedAmount*fee) / 100;
+        uint256 feeAmount = (seedAmount*fee) / PRECISION;
+
+        // seed amount vested per second > zero, i.e. amountVestedPerSecond = seedAmount/vestingDuration
+        require(
+            seedAmount >= vestingDuration,
+            "Seed: amountVestedPerSecond > 0");
 
         // total fundingAmount should not be greater than the hardCap
-        require( fundingBalance + _fundingAmount <= hardCap,
+        require( fundingCollected + _fundingAmount <= hardCap,
             "Seed: amount exceeds contract sale hardCap");
 
-        require( seedRemainder >= seedAmount,
-            "Seed: seed distribution would be exceeded");
-
-        fundingCollected = fundingBalance + _fundingAmount;
+        fundingCollected += _fundingAmount;
 
         // the amount of seed tokens still to be distributed
         seedRemainder -= seedAmount;
         feeRemainder  -= feeAmount;
 
-        // Here we are sending amount of tokens to pay for seed tokens to purchase
-        require(fundingToken.transferFrom(msg.sender, address(this), _fundingAmount), "Seed: no tokens");
-
         if (fundingCollected >= softCap) {
             minimumReached = true;
         }
         if (fundingCollected >= hardCap) {
             maximumReached = true;
-            vestingStartTime = _currentTime();
+            vestingStartTime = block.timestamp;
         }
 
-        _addFunder(
-            msg.sender,
-            (funders[msg.sender].seedAmount + seedAmount),         // Previous Seed Amount + new seed amount
-            (funders[msg.sender].fundingAmount + _fundingAmount),  // Previous Funding Amount + new funding amount
-            funders[msg.sender].totalClaimed,
-            (funders[msg.sender].fee + feeAmount),                  // Previous Fee + new fee
-            funders[msg.sender].feeClaimed
-            );
+        //functionality of addFunder
+        if (funders[msg.sender].fundingAmount==0) {
+            totalFunderCount++;
+        }
+        // totalClaimed is always going to be zero while buying
+        funders[msg.sender].fundingAmount += _fundingAmount;
+
+        // Here we are sending amount of tokens to pay for seed tokens to purchase
+        require(
+            fundingToken.transferFrom(msg.sender, address(this), _fundingAmount),
+            "Seed: funding token transferFrom failed"
+        );
 
         // buyer, seed token purchased in this transaction (not the total amount of seed purchased)
```

Comment:
also a confusing comment, not sure why do we have to write what value is not in the event

Reply:
Removed